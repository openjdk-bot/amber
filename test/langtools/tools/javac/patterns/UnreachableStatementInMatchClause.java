/*
 * @test /nodynamiccopyright/
 * @summary Ensure that unreachable statements in match clauses are detected corrected.
 * @compile/fail/ref=UnreachableStatementInMatchClause.out -XDrawDiagnostics UnreachableStatementInMatchClause.java
 */
public class UnreachableStatementInMatchClause {
    public static void main(String[] args) {
        Object o = "Hello world";
        switch (o) {
            case String s:
                System.out.println(s);
                break;
                throw new AssertionError("oops");
            default:
                System.out.println("hmm");
                break;
                throw new AssertionError("oops once more");
        }
    }
}
