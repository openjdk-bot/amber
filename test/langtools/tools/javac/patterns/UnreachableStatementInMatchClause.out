UnreachableStatementInMatchClause.java:13:17: compiler.err.unreachable.stmt
UnreachableStatementInMatchClause.java:17:17: compiler.err.unreachable.stmt
2 errors
